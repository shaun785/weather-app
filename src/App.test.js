import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import App from './App';

jest.mock('./api/fetchWeather');

describe('check search input', () => {
  it ('render header', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});


describe('test search input', () => {
  it ('render header', async () => {
    const wrapper = mount(<App />);
    wrapper.find('input').at(0).simulate('change', { target: { value: 'sydney' } });

    wrapper.find('input').at(0).simulate('keypress', {key: 'Enter'})

    await act(async () => {
      await new Promise(resolve => setTimeout(resolve)).then(() => wrapper.update());
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
