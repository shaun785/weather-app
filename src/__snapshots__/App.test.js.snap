// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check search input render header 1`] = `"<div class=\\"main-container\\"><input type=\\"text\\" class=\\"search\\" placeholder=\\"search city...\\" value=\\"\\"/></div>"`;

exports[`test search input render header 1`] = `"<div class=\\"main-container\\"><input type=\\"text\\" class=\\"search\\" placeholder=\\"search city...\\" value=\\"\\"><div class=\\"city\\"><h2 class=\\"city-name\\"><span>Sydney</span><sup>Australia</sup></h2><div class=\\"city-temp\\">30<sup>Â°C</sup></div><div class=\\"info\\"><img class=\\"city-icon\\" src=\\"https://openweathermap.org/img/wn/Cloudy@2x.png\\" alt=\\"Cloudy\\"><p>Cloudy</p></div></div></div>"`;
